lib/collection/src/shards/replica_set/update.rs

```diff
@@ -136,6 +136,25 @@ impl ShardReplicaSet {
         operation: CollectionUpdateOperations,
         wait: bool,
     ) -> CollectionResult<UpdateResult> {
+        // `ShardRepilcaSet::update_impl` is not cancel safe, so this method is not cancel safe.
+
+        loop {
+            if let Some(res) = self.update_impl(operation.clone(), wait).await? {
+                return Ok(res);
+            }
+
+            log::warn!("Operation {operation:?} was rejected by some node(s), retrying...");
+        }
```

Comment:
Some notes:
- I just realized I did not implement it exactly right: I release the clock between `update_impl` calls. Will fix this in a moment. 😂
- This spins and retries the operation infinitely, we probably want to limit the number of retries.
- The whole `update` is marked as not cancel safe, even though we should be safe to cancel "between" the calls to `update_impl`...
  - ...but I'm not sure if we can easily make it "cancellable between `update_impl` calls", so it might still spin in the background for a while, even if client drops the request...
  - ...which should be OK, once we limit the number of retries

Reply:
So, the question is: how many retries do we want? @generall @timvisee